app/src/controllers/authenticationController.js

```diff
@@ -66,13 +80,6 @@ export default class AuthenticationController {
     const certPath = process.env.SAML_CERT_PATH || "./certs/cert.pem";
     winston.log("info", "Reading SAML certificate file from %s", certPath);
 
-    const metadata = spidStrategy.generateServiceProviderMetadata(
-      fs.readFileSync(certPath, "utf-8")
-    );
-
-    res
-      .status(200)
-      .set("Content-Type", "application/xml")
-      .send(metadata);
+    getMetadataHandler(certPath, spidStrategy)(req, res);
```

Comment:
wait sorry I didn't see that this was inside a class, my suggestion to get the actual certificate content and the `spidStrategy` in the controller

reading of the certificate file should happen outside the handler and during server init (i.e. before you instantiate `acsController`)

Reply:
Ok, I've refactored the file reading